### DISCUSSION TITLE
Binaries in GitHub Releases, M1 macOS

### DISCUSSION THREAD

**Discussion by cyrusmsk** [Superbelko/ohmygentool#26] [Ideas]
Will it be possible to publish binaries for popular combinations?
like x86_64 Ubuntu, arm64 macOS?

So users won't need to build LLVM by themself. Like LDC making Releases of their compiler.

Also documentation mentioned that supported versions of LLVM till 11. However, macOS already has 14. And even LDC supported LLVM 14 (15 and 16 WIP). Does this lib work on newer versions of LLVM?

Can anyone confirm work on M1 macOS?

**Comment by Superbelko** [comment]
Hi, unfortunately it will likely won't work with LLVM 12 or later, keeping the development up to date takes up way too many time than I currently have, I might see if it compiles when I get to my ubuntu machine.

Another problem is that building a debug version takes up roughly 70 GB of disk space both for LLVM and Clang (so 140 GB in total), and I always have disk space issues. Developing without debug info is basically impossible, it is impossible to tell how to use AST nodes or what they could do or other useful properties.

Now about macOS:
If Apple provides dev versions of LLVM/Clang than you could try build this tool yourself, though since nobody reported if it actually works on Mac, let alone on ARM chips. 
For that you will need to update libraries and linker flags, no idea if it uses plain static libs and shared libs or a framework but D can do both just fine.

In the meantime you could try using docker image, however when I tried it on Windows because the itself was built in Linux environment it was generating bindings for Linux target which obviously won't work in less than trivial scenarios.

As for shipping binaries - again, the problem is that this will require enormous resources to build the LLVM and Clang and this is unlikely doable with github actions.
Another problem is like I've mentioned about docker is that the tool uses clang standard headers location which not always matches the installed compiler.


Summary:
- non local builds isn't quite tested, so it may fail to find system headers
- target triple configuration doesn't seems to work (e.g. docker image ignores triple target)
- on Linux systems it will have shared library search paths issues (no RPATH, must set correct LD_LIBRARY_PATH manually)
- it is kind of stale now because I have limited disk space and no free time
- shipping binaries isn't there except Windows
- c++17 or above support is lacking, especially compile-time and templates features

Other than that both D (ldc2) and LLVM/Clang (with modules) should work just fine.

**Comment by Superbelko** [comment]
I've updated the tool to support clang14, currently in develop branch, will merge later if CI runs ok.
